build-system/babel-plugins/babel-plugin-amp-mode-transformer/index.js

```diff
@@ -24,13 +24,25 @@ const {dirname, join, relative, resolve} = require('path').posix;
 // This plugin is only executed when bundling for production builds (minified).
 module.exports = function ({types: t}) {
   let getModeFound = false;
+  let modeNamespaceFound = false;
   return {
     visitor: {
       ImportDeclaration({node}, state) {
         const {source, specifiers} = node;
         if (!source.value.endsWith('/mode')) {
           return;
         }
+
+        if (source.value === '#core/mode') {
+          if (
+            specifiers.length === 1 &&
+            specifiers[0].local.name === 'mode' &&
+            specifiers[0].type === 'ImportNamespaceSpecifier'
+          ) {
+            modeNamespaceFound = true;
```

Comment:
This would need to be reset during a `Program` enter. But it's entirely unneeded, just use `referencesImport`.

Reply:
Whoah.